tests/manage/pv_services/pv_encryption/test_vault_namespace_override_using_tenant_configmap.py

```diff
@@ -0,0 +1,195 @@
+        ),
+        pytest.param(
+            "v2", kmsprovider, False, marks=pytest.mark.polarion_id("OCS-4640")
+        ),
+    ]
+
+
+@pytest.mark.parametrize(
+    argnames=argnames,
+    argvalues=argvalues,
+)
+@tier1
+@skipif_ocs_version("<4.10")
+@bugzilla("2050056")
+class TestEncryptedRbdTenantConfigmapOverride(ManageTest):
+    """
+    Tests to check Tenant configmap override vault namespace or not
+    1. Create a new namespace in OCP and create the ceph-csi-kms-token secret
+    2. Create an encryption enabled storageclass using vault namespaces
+    3. Create a configmap in the tenant namespace to override the vault namespace
+    4. Update the ceph-csi-kms-token secret in the tenant namespace to provide access to the new namespace and backend
+       path
+    5. Create a new PVC
+    6. Verify that the PVC is bound and the key is created in the specified path in Vault
+    7. Mounting the PVC in an app pod and running IO
+
+    """
+
+    @pytest.fixture()
+    def setup(
+        self,
+        kv_version,
+        use_vault_namespace,
+        pv_encryption_kms_setup_factory,
+        project_factory,
+        storageclass_factory,
+    ):
+        """
+        Setup csi-kms-connection-details configmap
+        """
+
+        log.info("Setting up csi-kms-connection-details configmap")
+        self.kms_vault = pv_encryption_kms_setup_factory(
+            kv_version, use_vault_namespace
+        )
+        log.info("csi-kms-connection-details setup successful")
+        # Create a project
+        self.proj_obj = project_factory()
+
+        # Create an encryption enabled storage class for RBD
+        self.sc_obj = storageclass_factory(
+            interface=constants.CEPHBLOCKPOOL,
+            encrypted=True,
+            encryption_kms_id=self.kms_vault.kmsid,
+        )
+
+    def test_encryptedrbd_pvc_status_with_tenant_configmap_override(
+        self,
+        request,
+        setup,
+        multi_pvc_factory,
+        kv_version,
+        kms_provider,
+        pod_factory,
+        use_vault_namespace,
+    ):
+
+        self.kms_obj = kms.Vault()
+        vault_resource_name = create_unique_resource_name("test", "vault")
+        if use_vault_namespace:
+            self.kms_obj.vault_create_namespace(namespace=vault_resource_name)
+        self.kms_obj.vault_create_backend_path(
+            backend_path=vault_resource_name, kv_version=kv_version
+        )
+        self.kms_obj.vault_create_policy(policy_name=vault_resource_name)
+
+        def kms_obj_cleanup():
+            self.kms_obj.remove_vault_backend_path(
+                vault_namespace=self.kms_obj.vault_namespace
+            )
+            self.kms_obj.remove_vault_policy(
+                vault_namespace=self.kms_obj.vault_namespace
+            )
+            if self.kms_obj.vault_namespace:
+                self.kms_obj.remove_vault_namespace()
+
+        request.addfinalizer(kms_obj_cleanup)
```

Comment:
should be done within the test setup so the cleanup will take place in case of an earlier phase test or setup failure